app/code/Magento/Catalog/Model/ImageUploader.php

```diff
@@ -46,132 +47,43 @@ class ImageUploader
 
     /**
-     * Allowed extensions
-     *
-     * @var string
-     */
-    protected $allowedExtensions;
-
-    /**
-     * List of allowed image mime types
-     *
-     * @var string[]
-     */
-    private $allowedMimeTypes;
-
-    /**
-     * ImageUploader constructor
+     * ImageUploader constructor.
      *
      * @param \Magento\MediaStorage\Helper\File\Storage\Database $coreFileStorageDatabase
      * @param \Magento\Framework\Filesystem $filesystem
      * @param \Magento\MediaStorage\Model\File\UploaderFactory $uploaderFactory
      * @param \Magento\Store\Model\StoreManagerInterface $storeManager
      * @param \Psr\Log\LoggerInterface $logger
-     * @param string $baseTmpPath
-     * @param string $basePath
-     * @param string[] $allowedExtensions
-     * @param string[] $allowedMimeTypes
+     * @param Provider $provider
+     * @param Name $fileNameLookup
+     * @throws \Magento\Framework\Exception\FileSystemException
      */
     public function __construct(
         \Magento\MediaStorage\Helper\File\Storage\Database $coreFileStorageDatabase,
         \Magento\Framework\Filesystem $filesystem,
         \Magento\MediaStorage\Model\File\UploaderFactory $uploaderFactory,
         \Magento\Store\Model\StoreManagerInterface $storeManager,
         \Psr\Log\LoggerInterface $logger,
-        $baseTmpPath,
-        $basePath,
-        $allowedExtensions,
-        $allowedMimeTypes = []
+        Provider $provider,
+        Name $fileNameLookup
     ) {
         $this->coreFileStorageDatabase = $coreFileStorageDatabase;
         $this->mediaDirectory = $filesystem->getDirectoryWrite(\Magento\Framework\App\Filesystem\DirectoryList::MEDIA);
         $this->uploaderFactory = $uploaderFactory;
         $this->storeManager = $storeManager;
         $this->logger = $logger;
-        $this->baseTmpPath = $baseTmpPath;
-        $this->basePath = $basePath;
-        $this->allowedExtensions = $allowedExtensions;
-        $this->allowedMimeTypes = $allowedMimeTypes;
-    }
-
-    /**
-     * Set base tmp path
-     *
-     * @param string $baseTmpPath
-     *
-     * @return void
-     */
-    public function setBaseTmpPath($baseTmpPath)
-    {
-        $this->baseTmpPath = $baseTmpPath;
-    }
-
-    /**
-     * Set base path
-     *
-     * @param string $basePath
-     *
-     * @return void
-     */
-    public function setBasePath($basePath)
-    {
-        $this->basePath = $basePath;
-    }
-
-    /**
-     * Set allowed extensions
-     *
-     * @param string[] $allowedExtensions
-     *
-     * @return void
-     */
-    public function setAllowedExtensions($allowedExtensions)
-    {
-        $this->allowedExtensions = $allowedExtensions;
-    }
-
-    /**
-     * Retrieve base tmp path
-     *
-     * @return string
-     */
-    public function getBaseTmpPath()
-    {
-        return $this->baseTmpPath;
-    }
-
-    /**
-     * Retrieve base path
-     *
-     * @return string
-     */
-    public function getBasePath()
```

Comment:
Is 2.4 a breaking change? Am I allowed to remove these? I could not see any usaged in place for them apart from in the class.